esmvaltool/diag_scripts/land_carbon_cycle/diag_zonal_correlation.py

```diff
@@ -0,0 +1,452 @@
+        r_tau_pr_c_tas_mod = r_mod[:, 1]
+        if row_mod in ['MultiModelMedian', 'MultiModelMean']:
+            sp1.plot(np.ma.masked_equal(r_tau_tas_c_pr_mod, np.nan),
+                     lats_mod.points,
+                     lw=1.1,
+                     color='blue',
+                     label=row_mod)
+            sp2.plot(np.ma.masked_equal(r_tau_pr_c_tas_mod, np.nan),
+                     lats_mod.points,
+                     lw=1.1,
+                     color='blue',
+                     label=row_mod)
+        else:
+            sp1.plot(np.ma.masked_equal(r_tau_tas_c_pr_mod, np.nan),
+                     lats_mod.points,
+                     lw=0.3,
+                     label=row_mod)
+            sp2.plot(np.ma.masked_equal(r_tau_pr_c_tas_mod, np.nan),
+                     lats_mod.points,
+                     lw=0.3,
+                     label=row_mod)
+
+    # normalized mean correlations from model
+
+    # remove the multimodel estimates
+    models = list(zonal_correlation_mod.keys())
+    for _mm in multiModels:
+        if _mm in models:
+            models.remove(_mm)
+
+    nmodels = len(models)
+
+    r_tau_pr_c_tas_all = np.ones((len(lats_obs.points), nmodels)) * np.nan
+    r_tau_tas_c_pr_all = np.ones((len(lats_obs.points), nmodels)) * np.nan
+    for row_m in range(nmodels):
+        row_mod = models[row_m]
+        r_mod = zonal_correlation_mod[row_mod]['data']
+        lats_mod = zonal_correlation_mod[row_mod]['latitude']
+        r_tau_tas_c_pr_all[:, row_m] = r_mod[:, 0]
+        r_tau_pr_c_tas_all[:, row_m] = r_mod[:, 1]
+
+    r_mmod, r_mmod_std_low, r_mmod_std_hi = _get_multimodel_stats(
+        r_tau_tas_c_pr_all)
+    sp1.plot(np.ma.masked_equal(r_mmod, np.nan),
+             lats_mod.points,
+             color='red',
+             ls='--',
+             lw=1,
+             label='Norm. Mean r')
+    sp1.fill_betweenx(lats_mod.points,
+                      np.ma.masked_equal(r_mmod_std_low, np.nan),
+                      np.ma.masked_equal(r_mmod_std_hi, np.nan),
+                      facecolor='#42d4f4',
+                      alpha=0.25)
+
+    r_mmod, r_mmod_std_low, r_mmod_std_hi = _get_multimodel_stats(
+        r_tau_pr_c_tas_all)
+
+    sp2.plot(np.ma.masked_equal(r_mmod, np.nan),
+             lats_mod.points,
+             color='red',
+             ls='--',
+             lw=1,
+             label='Norm. Mean r')
+    sp2.fill_betweenx(lats_mod.points,
+                      np.ma.masked_equal(r_mmod_std_low, np.nan),
+                      np.ma.masked_equal(r_mmod_std_hi, np.nan),
+                      facecolor='#42d4f4',
+                      alpha=0.25)
+
+    plt.gca().yaxis.set_label_position("right")
+
+    # draw the legend
+    leg = xu.draw_line_legend(ax_fs=fig_config['ax_fs'])
+
+    plt.savefig(plot_path,
+                bbox_inches='tight',
+                bbox_extra_artists=[leg],
+                dpi=450)
+    plt.close()
```

Comment:
since you are always saving with the same parameters, you could write a save function - that'll eliminate quite a few repetitions

Reply:
Moved it to plotUtils.